### PR TITLE
Update codeowners

### PR DESCRIPTION
**Pull request by swernli**
Adding a few more owners for folders in runtime.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -12,36 +12,36 @@
...
+/src/Simulation/EntryPointDriver/                      @samarsha
+/src/Simulation/EntryPointDriver.Tests/                @samarsha
 /src/Simulation/Native/                                @kuzminrobin
 /src/Simulation/NativeSparseSimulator/                 @kuzminrobin
 /src/Simulation/QCTraceSimulator/                      @DmitryVasilevsky
 /src/Simulation/QCTraceSimulator.Tests/                @DmitryVasilevsky
-/src/Simulation/qdk_sim_rs/
-/src/Simulation/QSharpCore/
-/src/Simulation/QSharpFoundation/
+/src/Simulation/qdk_sim_rs/                            @swernli
```
Comment (kuzminrobin) [inline .github/CODEOWNERS:21]: Consider
```suggestion
/src/Simulation/qdk_sim_rs/                            @swernli @cgranade
```
or
```suggestion
/src/Simulation/qdk_sim_rs/                            @cgranade @swernli
```

**Review by kuzminrobin — APPROVED** [review]

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -12,36 +12,36 @@
 /src/Simulation/AutoSubstitution.Tests/
 /src/Simulation/Common/                                @kuzminrobin
 /src/Simulation/Core/                                  @kuzminrobin
-/src/Simulation/EntryPointDriver/
-/src/Simulation/EntryPointDriver.Tests/
+/src/Simulation/EntryPointDriver/                      @samarsha
+/src/Simulation/EntryPointDriver.Tests/                @samarsha
```
Comment (cesarzc) [inline .github/CODEOWNERS:16]:
```suggestion
/src/Simulation/EntryPointDriver/                      @samarsha @cesarzc
/src/Simulation/EntryPointDriver.Tests/                @samarsha @cesarzc
```

**Comment by cesarzc** [comment]
@swernli can you please add me as owner of /src/Azure/?

**Review by cesarzc — APPROVED** [review]
Added some suggestions.

**Pull request merged**